### DISCUSSION TITLE
feature: background asset loading

### DISCUSSION THREAD

**Discussion by jcsnider** [AscensionGameDev/Intersect-Engine#2706] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently the client loads textures, audio files, and all other resources as needed, even if in the middle of rendering a frame. These loads impact fps because the game loop waits for the loads to complete. The fps hit can be pretty massive if games reside on standard hard drives

**Describe the solution you'd like**
Background loading of new assets as needed is a must.

**Additional context**

**Comment by Arufonsu** [comment]
This approach optimizes performance by loading everything at startup and caching it in memory. This way, we avoid reloading data on every tick, which is very inefficient.

[preload.patch](https://github.com/user-attachments/files/18097884/preload.patch)

I have been testing this on ASO (where it seems to behave better in terms of performance during the gameplay), We have been testing it while working with Myra as well. This approach will be merged into main at some point for sure, solving this feat request.
